GVFS/GVFS.Upgrader/UpgradeOrchestrator.cs

```diff
@@ -0,0 +1,375 @@
+                        this.output.WriteLine(cleanUpError);
+                        this.ExitCode = ReturnCode.Success;
+                    }
+                }
+            }
+            
+            if (this.ExitCode == ReturnCode.GenericError)
+            {
+                error = Environment.NewLine + "ERROR: " + error;
+                this.output.WriteLine(error);
+            }
+
+            this.output.WriteLine("Press Enter to exit.");
+            if (this.input == Console.In)
+            {
+                this.input.ReadLine();
+            }
+            
+            if (this.shouldExit)
+            {
+                Environment.Exit((int)this.ExitCode);
+            }
+        }
+
+        private bool LaunchInsideSpinner(Func<bool> method, string message)
+        {
+            return ConsoleHelper.ShowStatusWhileRunning(
+                method,
+                message,
+                this.output,
+                this.output == Console.Out && !GVFSPlatform.Instance.IsConsoleOutputRedirectedToFile(),
+                null);
+        }
+
+        private bool TryRunUpgradeInstall(out Version newVersion, out string error)
+        {
+            newVersion = null;
+
+            Version newGVFSVersion = null;
+            GitVersion newGitVersion = null;
+            string errorMessage = null;
+            if (!this.LaunchInsideSpinner(
+                () =>
+                {
+                    if (!this.TryCheckIfUpgradeAvailable(out newGVFSVersion, out errorMessage) ||
+                        !this.TryGetNewGitVersion(out newGitVersion, out errorMessage))
+                    {
+                        return false;
+                    }
+
+                    this.LogInstalledVersionInfo();
+                    this.LogVersionInfo(newGVFSVersion, newGitVersion, "Available Version");
+
+                    if (!this.preRunChecker.TryRunPreUpgradeChecks(newGitVersion, out errorMessage))
+                    {
+                        return false;
+                    }
+
+                    if (!this.TryDownloadUpgrade(newGVFSVersion, out errorMessage))
+                    {
+                        return false;
+                    }
+
+                    return true;
+                },
+                "Downloading"))
+            {
+                error = errorMessage;
+                return false;
+            }
+
+            if (!this.LaunchInsideSpinner(
+                () =>
+                {
+                    if (!this.preRunChecker.TryUnmountAllGVFSRepos(out errorMessage))
+                    {
+                        return false;
+                    }
+
+                    this.remount = true;
+
+                    if (!this.TryInstallGitUpgrade(newGitVersion, out errorMessage))
+                    {
+                        return false;
+                    }
+
+                    return true;
+                },
+                "Installing Git"))

```

Comment:
I would include the version that we're installing here. It's mildly interesting to users, but will also be helpful to us when users send us screenshots during support issues.